Fix left button in header [Android]

grabbou opened the pull request:
Fixes #432 

For some reason, `onLayoutIOS` was missing Platform check.

ayalpani commented:
Thanks, I npm updated but this didn't fix the issue for me.

ayalpani commented:
But maybe my case is different. I am adding back button through navigationOptions:

```
  static navigationOptions = {
    title: 'Details',

    header: ({ goBack }) => {
      return {
        visible: true,
        left: <BackButton onPress={ () => goBack() } />
      }
    },
  }
```

grabbou commented:
Not sure what is your case as I don't see your comment in the original issue. What exactly is broken for you?

ayalpani commented:
hi @grabbou 

In my case, it also looks like this, the arrow is covered in parts:

![](https://cloud.githubusercontent.com/assets/2179670/23189219/f184404c-f8b7-11e6-97a4-151368944333.png)

grabbou commented:
Make sure you are running latest master (what is your react-navigation dependency in package.json)? Remove node_modules and reinstall. You likely have a stale version.

Also, I don't understand this part: `left: <BackButton onPress={ () => goBack() } />`. Where is that component coming from? After removing it, does the problem occur?

ayalpani commented:
I had checked the dependency before. This is from my package.json:

`"react-navigation": "git+https://github.com/react-community/react-navigation/",`

Anyway, I deleted node_modules but nothing changed.

This is where I got the BackButton:

```
import { StackNavigator, CardStack }     from 'react-navigation'
const { BackButton } = CardStack.Header
```

I am trying to manually add it to the header, so if I remove the component, there will be no back button at all ...

grabbou commented:
Makes sense.

Anyways, I've added:

```js
MyHomeScreen.navigationOptions = {
  title: 'Welcome',
  header: {
    left: <CardStack.Header.BackButton />,
  },
};
```

to the SimpleStack.js file in examples/NavigationPlayground, just like you
did and still, I don't see the glitch.

Be sure to restart the packager as well.

On Wed, 22 Feb 2017 at 16:33 ayalpani <notifications@github.com> wrote:

> I had checked the dependency before. This is from my package.json:
>
> "react-navigation": "git+
> https://github.com/react-community/react-navigation/",
>
> Anyway, I deleted node_modules but nothing changed.
>
> This is where I got the BackButton:
>
> import { StackNavigator, CardStack }     from 'react-navigation'
> const { BackButton } = CardStack.Header
>
> I am trying to manually add it to the header, so if I remove the
> component, there will be no back button at all ...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/react-community/react-navigation/pull/437#issuecomment-281703883>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACWcxgGUl8Dv8IMq2tDLsE5x2XLrU_Isks5rfFVcgaJpZM4MIhZe>
> .
>

ayalpani commented:
Yes, I tried what you tried too and it works perfectly. I will further investigate and let you know when I have found the issue. Thanks for your time
